ext/ffi/ffi.c

```diff
@@ -560,22 +576,22 @@ static zend_always_inline void zend_ffi_cdata_to_zval(zend_ffi_cdata *cdata, voi
 				return;
 #endif
 			case ZEND_FFI_TYPE_UINT8:
-				ZVAL_LONG(rv, *(uint8_t*)ptr);
+				ZVAL_LONG(rv, ZEND_FFI_READ_NARROW(uint8_t, ptr, is_ret));
```

Comment:
Instead of ``zend_ffi_cdata_to_zval()`` modification, we should pass the "proper" type at the first place (in ``ffi_trampoline``).

Reply:
I committed this separately to this PR.
I was able to get rid of the `ZEND_FFI_READ_NARROW` macros, but I needed to keep the special handling of `ZEND_FFI_TYPE_BOOL` to use `ZVAL_BOOL` and `ZEND_FFI_TYPE_CHAR` with `ZVAL_CHAR`. So therefore I kept the return handling in this function. I'm not sure this is much simpler, I preferred the old way to be honest.